lib/asserts.nix

```
@@ -2,78 +2,122 @@
 
 rec {
 
-  /* Throw if pred is false, else return pred.
-     Intended to be used to augment asserts with helpful error messages.
+  /**
+    Throw if pred is false, else return pred.
+    Intended to be used to augment asserts with helpful error messages.
 
-     Example:
-       assertMsg false "nope"
-       stderr> error: nope
+    # Example
```

```suggestion
    # Examples
```

Can you move examples to be the last header for each function? Since examples can get kinda long, it's better to keep the rest of the information closer.
(applies to other functions too)

---

Changed. Though it feels kind of strange to me now.
I guess this is a tradeoff, either having a concrete usage example first, rather than some abstract `type` and `inputs` gibberish.

---

This is whay i considered starting with examples because it lowers the entry barrier:
> Beginners or developers looking for quick solutions tend to prefer examples upfront. It helps them understand how the function works in a real-world scenario and can be particularly useful for those who learn by doing.

---

I consider the `lib.asserts` case to be simple because each function only needs a single paragraph of explanation. Usually you'll have multiple paragraphs to explain what something does, and my approach has been to directly link to examples from that text (see [this example](https://nixos.org/manual/nixpkgs/unstable/#ssec-pkgs-dockerTools-streamLayeredImage)). For beginners who are looking for quick solutions, they can read the explainer and follow to the example they're more interested in, because usually the function will have enough options that a single example doesn't cover all use cases as well.

---

Looked a little bit into other documentation websites and agree that its more common to have examples at the bottom.